Stack/Opc.Ua.Core/Security/Certificates/X509CertificateStore.cs

```diff
@@ -145,19 +146,34 @@ public Task Add(X509Certificate2 certificate, string password = null)
                         (Environment.OSVersion.Platform == PlatformID.Win32NT))
                     {
                         // see https://github.com/dotnet/runtime/issues/29144
-                        var temp = X509Utils.GeneratePasscode();
-                        using (var persistable = new X509Certificate2(certificate.Export(X509ContentType.Pfx, temp), temp,
+                        var tempPass = X509Utils.GeneratePasscode();
+                        //ensure we control the temp file creation and deletion.
+                        // If you import a byte array a temp file is created and that may not be deleted.
+                        var tempFileName = Path.GetTempFileName();
+                        var tempCertificate = certificate.Export(X509ContentType.Pfx, tempPass);
+
+                        using (var tempFile = new FileStream(tempFileName, FileMode.Open))
+                        {
+                            tempFile.Write(tempCertificate,0, tempCertificate.Length);
+                        }
+
+                        using (var persistable = new X509Certificate2(tempFileName, tempPass,
                             X509KeyStorageFlags.PersistKeySet))
                         {
                             store.Add(persistable);
                         }
+                        File.Delete(tempFileName);
                     }
                     else
 #endif
                     if (certificate.HasPrivateKey && m_noPrivateKeys)
                     {
                         // ensure no private key is added to store
-                        store.Add(new X509Certificate2(certificate.RawData));
+                        // ensure the interim certificate is disposed after use
+                        using (X509Certificate2 cert = new X509Certificate2(certificate.RawData))
```

Comment:
or create interim cert only if there is no private key in the cert. I believe dir store has now also a check, so this extra cert generation may be obsolete.